packages/belos/src/BelosTsqrOrthoManagerImpl.hpp

```diff
@@ -128,9 +128,6 @@ namespace Belos {
   /// yourself.
   template<class Scalar>
   class ReorthogonalizationCallback :
```

Comment:
```suggestion
  class ReorthogonalizationCallback
```

The colon needs to be dropped

Reply:
I am likewise concerned that changes made in this PR are not being thoroughly tested and may result in errors when optional code is enabled by the user.

Reply:
Sorry, I'm not a trilinos developer, how can I test superlu_dist?

Reply:
@cjobrie you'll need an install of superlu_dist and metis available, and you might need to point to them in the cmake configuration, something like this (where you would need to provide the relevant env variable paths to the tpl installs):

```
-DTPL_ENABLE_SuperLUDist:BOOL=ON \ -DSuperLUDist_INCLUDE_DIRS:PATH="${SUPERLU_DIST_INCLUDE_PATH}" \
-DSuperLUDist_LIBRARY_DIRS:PATH="${SUPERLU_DIST_LIBRARY_PATH};${METIS_LIBRARY_PATH}/libmetis.a;${BLAS_LIBS};${LAPACK_LIBS};gfortran;gomp;m" \
# The gfortran;gomp;m additions may be unnecessary, but some/all may help if you encounter linker errors
```

With Amesos2 you can enable SuperLU_Dist solver with `-DAmesos2_ENABLE_SuperLUDist:BOOL=ON`

If you're a spack user there may be formula you can use as well, though I'm not familiar enough to provide details

Reply:
I'm sorry, Trilinos passed all ctest tests on intel, but I can't quite get `superlu` to build. I've no idea how to do it and the instructions don't tell me where the libraries could be installed. 

Using the `-DfAmesos2...` command errors out as reported below:

```console
CMake Error at cmake/tribits/core/package_arch/TribitsProcessEnabledTpls.cmake:278 (message):
  ERROR: TPL_SuperLUDist_NOT_FOUND=TRUE, aborting!
```

Reply:
The only `libmetis.a` I've found is at "projects/cde/v3/cee/spack/opt/spack/linux-rhel7-x86_64/intel-2021.1.2/metis-5.1.0-vxnfwinsazp37pgcjusoa3xuvhxvdiiq/lib/libmetis.a" However, using your suggestions, I still can't seems to get Amesos to recognize it. 

```concole
CMake Error at packages/amesos/CMakeLists.txt:16 (MESSAGE):
  The Amesos support for the SuperLUDist TPL requires the ParMETIS TPL.
  Either disable Amesos SuperLUDist support or enable the ParMETIS TPL.
```

Reply:
@cjobrie that is picking up Amesos instead of Amesos2, since these changes only impact Amesos2 I'd suggest disabling Amesos (-DTrilinos_ENABLE_Amesos=OFF).
It looks like ParMetis is required in addition to Metis (I don't build with superlu_dist often, had forgotten about this).
You'll need to add parmetis in your spack configuration, and you may need additional config options like:
```
-D TPL_ENABLE_METIS:BOOL=ON \
  -D METIS_INCLUDE_DIRS:PATH="${METIS_PATH}/include" \
  -D METIS_LIBRARY_DIRS:PATH="${METIS_PATH}/lib" \
  -D METIS_LIBRARY_NAMES:STRING="metis" \
-D TPL_ENABLE_ParMETIS:BOOL=ON \
  -D ParMETIS_INCLUDE_DIRS:PATH="${PARMETIS_PATH}/include;${METIS_PATH}/include" \
  -D ParMETIS_LIBRARY_DIRS:PATH="${PARMETIS_PATH}/lib" \
  -D ParMETIS_LIBRARY_NAMES:STRING="parmetis" \
...
  -D SuperLUDist_LIBRARY_DIRS:PATH="${SUPERLU_DIST_PATH}/lib/libsuperlu_dist.a;${METIS_PATH}/lib/libmetis.a;${PARMETIS_PATH}/lib/libparmetis.a;${BLAS_PATH}/lib/libopenblas.a" \
```

Sorry I can't provide more specific guidance rather than configuration snips, using TPLs can be finicky (though spack should help with this).

I'm trying to build superlu_dist and hit some bumps myself. Let me know if any of the above is useful enough to get a complete configuration and build on your end, I'll keep working at it as well

Reply:
I added the pretest inspection label so we can check the AT results

Reply:
Unfortunately, this suggestion breaks everything. Is there anyone who has this setup to try this. I don't have the time or expertise to fix this kind of detailed  problem with a build system I don't have any experience with.


```CMake Error at cmake/tribits/core/utils/CombinedOption.cmake:129 (message):
  Error: you can not enable the option TEKO_HAVE_EPETRA unless you also
  enable the options Teko_ENABLE_Epetra, Teko_ENABLE_ThyraEpetraAdapters,
  Teko_ENABLE_ThyraEpetraExtAdapters, Teko_ENABLE_AztecOO,
  Teko_ENABLE_Ifpack, Teko_ENABLE_Amesos, Teko_ENABLE_ML.  The current option
  values are:

    Teko_ENABLE_Epetra='ON'
    Teko_ENABLE_ThyraEpetraAdapters='ON'
    Teko_ENABLE_ThyraEpetraExtAdapters='ON'
    Teko_ENABLE_AztecOO='ON'
    Teko_ENABLE_Ifpack='ON'
    Teko_ENABLE_Amesos='OFF'
    Teko_ENABLE_ML='ON'

Call Stack (most recent call first):
  packages/teko/CMakeLists.txt:25 (COMBINED_OPTION)
```

Reply:
@cjobrie yuck... I didn't know so much stuff was turned on. Alright, if you can try one case, reenable Amesos and also enable Parmetis. Reenabling Amesos should resolve the new stuff you just posted, and hopefully enabling Parmetis is sufficient to work out the dependency requested by SuperLU_Dist (this was a couple messages ago)

Reply:
Once again, I've tried:
```cmake -DTrilinos_ENABLE_Amesos=ON -DAmesos2_ENABLE_SuperLUDist:BOOL=ON -DTPL_ENABLE_METIS:BOOL=ON -DTPL_ENABLE_ParMETIS:BOOL=ON -DParMETIS_LIBRARY_DIRS:PATH="/projects/cde/v3/cee/spack/opt/spack/linux-rhel7-x86_64/intel-2021.1.2/parmetis-4.0.3-yj2npclamik5tmk2bj7l2asttqrz3pmc/lib" -DParMETIS_INCLUDE_DIRS:PATH="/projects/cde/v3/cee/spack/opt/spack/linux-rhel7-x86_64/intel-2021.1.2/parmetis-4.0.3-yj2npclamik5tmk2bj7l2asttqrz3pmc/include"  -C ../Trilinos-fork/genconfig.cmake ../Trilinos-fork 
```
and ParMETIS can't be found!!!